the-collab-lab/tcl-46-smart-shopping-list
# Add `npm test` to GH Actions

## Description

Added `npm test` to the Github Actions CI/CD environment. When we make a new push to `main` or `development`, this test will run to check if our `npm test` unit tests all work.

## Related Issue

Closes #31 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

![image](https://user-images.githubusercontent.com/77255525/184789337-bc270150-9755-4c20-a627-b466365bf991.png)
### After

Added a new workflow that runs `npm run test` for us :)
See : "Vitest Unit Tests"

![image](https://user-images.githubusercontent.com/77255525/184926274-df9b83ff-a079-4348-8f80-5d65e626a36f.png)

## Testing Steps / QA Criteria

Click `Checks` on above tab to see the added `Vitest Unit Tests` test. 
Pushed the rebase in order to update this testing view. All are passing now : )
![image](https://user-images.githubusercontent.com/77255525/184790368-06f50a25-42b6-42e9-b773-bd1626e5f9cd.png)

![image](https://user-images.githubusercontent.com/77255525/184926362-4cd21b0f-daf4-4289-a949-7a0d6c97ddb9.png)
